### ISSUE TITLE
add join support

### ISSUE THREAD

**Issue by siddontang** [pingcap/pd#215]
Now adding a member dynamically is not easy, see https://coreos.com/etcd/docs/latest/runtime-configuration.html
And we have to another supply initital_cluster_state flag too, I think we can use another way to do it. 

```
# start first pd
pd-server -name pd1 -host pd1

# start second pd and join the cluster
pd-server -name pd2 -host pd2 -join "pd1=http://pd1:2379"

# start third pd and joint the cluster
pd-server -name pd3 -host pd3 -join "pd1=http://pd1:2379"
# or
pd-server -name pd3 -host pd3 -join "pd2=http://pd2:2379"
# or
pd-server -name pd3 -host pd3 -join "pd1=http://pd1:2379,pd2=http://pd2:2379"
```

We start first pd and then use `join` mechanism to let following pd to join the cluster. The join arg can contain any valid pds in current cluster

/cc @disksing  @qiuyesuifeng @iamxy

**Comment by c4pt0r** [comment]
@overvenus  PTAL

**Comment by c4pt0r** [comment]
feel free to ask @siddontang if u have any question.

**Comment by overvenus** [comment]
Got it!

**Comment by siddontang** [comment]
you must notice that the join pd list must be `pd:2379`, not 2380, because 2380 is for internal peer communication, but here we need let pd  to add us in cluster

**Comment by xiang90** [comment]
This is pretty dangerous. For etcd, we intentionally make it tedious by requiring two steps + full expected configuration... Or people will make mistakes and screw up their clusters (we use join in the early version of etcd... we see too many troubles...).

**Comment by ngaut** [comment]
Thank you. @xiang90

**Comment by siddontang** [comment]
Hi @xiang90 

We also use two-step in join, it is the same as etcd add member doc says. First sending Add member request to etcd, then start new member again with corresponding args. 

Of course, this may be still dangerous even we add enough check.

@c4pt0r

**Comment by overvenus** [comment]
Fixed, #226
